src/main/src/main/java/org/geoserver/util/EntityResolverProvider.java

```diff
@@ -23,6 +28,54 @@
     public static final EntityResolverProvider RESOLVE_DISABLED_PROVIDER = new EntityResolverProvider(
             null);
 
+    /**
+     * In IDEs during development GeoTools sources can be in the classpath of GeoServer tests, this
+     * resolver allows them to be resolved while blocking the rest
+     */
+    public static final EntityResolver RESOLVE_DISABLED_PROVIDER_DEVMODE = new PreventLocalEntityResolver() {
```

Comment:
This seems very test-specific behaviour. Is there a way it could live in the `test/` side without (big) rework on the `main/` side? If not, could it be made more general so that you can just pass in a map of local locations => schemas?

Reply:
As you saw in other comments this is the indeed third iteration of this patch, and the least intrusive one of the three variations.
I've first stared by allowing all entities in the test, which required a big rework in existing tests checking external entities.

Then tried to only allow schema entities (anything ending with .xsd), that also required significant rework as various tests explicitly check for entity expansions on stuff ending with .xsd.

This version is the one that requires the least changes for a case that's indeed quite specific (developer working in Maven aware IDE with both GeoTools and GeoServer sources loaded).

Passing "paths" is more work and would not help, what would be needed are regular expressions matching paths, and eventually a different one for systemId and baseURI. It would also require to make changes down in Geotools.

If test specific code in this class is bothering you maybe the code could be reworked to have a DEFAULT_RESOLVER static non final field that defaults to ``PreventLocalEntityResolver.INSTANCE`` but that test code could replace with its own version, and move the test specific code to SystemTestSupport. 
Would that be better? Is it something you're interested in doing (a pull request can be made against this one, targetting the right branch on my clone)